Finsished Movement Script

agonzal1 opened the pull request:
movement + rudimentary collision test (added box colliders and updated rigidbody to dynamic)

agonzal1 committed:
Movement Script Boiler Plate

Boiler Plate code

agonzal1 committed:
Finished Basic Movement Script

Finished Basic Movement Script

agonzal1 committed:
Basic collision test

Collisions

agonzal1 committed:
Added Pause Menu

Implemented basic Pause Menu

agonzal1 commented:
Also Added Pause Menu

RylannAnd commented on Assets/Scripts/movement.cs:
```diff
@@ -22,6 +28,69 @@ void Start()
     void Update()
     {
-        
+        if(!PauseMenu.isPaused){
```
Do we need to reupdate after game is un-paused? This is a gap in my unity knowledge so let me know if we don't please!

Sren002 replied:
Yes, but that is handled in the PauseMenu script. This code just prevents movement inputs from doing anything while the game is paused, which could allow the player to still press wasd and move, even though game is paused.

RylannAnd reviewed (commented):
Left a couple of questions as I need a bit of help understanding!

Sren002 commented on Assets/Scripts/movement.cs:
```diff
@@ -22,6 +28,69 @@ void Start()
+        }
+    }
+
+    void HandleSpriteFlip() {
```
Vertical movement is handled by south and north sprites, i.e there is no need to flip the sprites because they already exist. This approach is done assuming that we only have east sprites, which is technically not true, but we can save memory by not having west sprites.

RylannAnd reviewed (approved):
Makes sense now!

agonzal1 committed:
Dust when changing directions

Created particle system that triggers when player changes directions. Also renamed player, and added it to player layer.

agonzal1 committed:
Created Main Menu

Created scene for main menu. Used sceneManager to swap from main menu to game scene.

agonzal1 committed:
Added Sound Manager

Added logic for sound manager and updating volume preferences.

agonzal1 committed:
Fixed Pause Menu

Pause Menu adequately displays and send to main menu.

agonzal1 committed:
Updated Resolution and Added Options Menu to Sample Scene

agonzal1 committed:
Added Pause Menu Functionality to Coffee Shop

agonzal1 committed:
Scrollable Component Experimentation

Created scrollable component that will be used for the menu where players can buy their drinks. Currently scrolls and is easily updatable with more drinks, however the purchasing functionality or the actual logic behind the shop has not been implemented.

agonzal1 committed:
Added some Art

agonzal1 committed:
Added basic Timer Functionality

Menu button now enables timer, see script comments for expected future behaviour.

Should also make it look nicer

agonzal1 committed:
Added Mocha to Menu

agonzal1 reviewed (commented):
Lovely

agonzal1 committed:
Different Drinks now Triggers Different Timers

Start Timer now takes an int which can be passed by the button press.

Also added some art and fixed panel anchoring.
